src/DynamoCoreWpf/Views/Preview/Watch3DView.xaml.cs

```diff
@@ -368,19 +404,147 @@ private void OnViewLoaded(object sender, RoutedEventArgs e)
             vm.ViewModel.Model.Logger.Log(string.Format("RENDER : Pixel Shader 3 Supported: {0}", pixelShader3Supported), LogLevel.File);
             vm.ViewModel.Model.Logger.Log(string.Format("RENDER : Pixel Shader 4 Supported: {0}", pixelShader4Supported), LogLevel.File);
             vm.ViewModel.Model.Logger.Log(string.Format("RENDER : Software Effect Rendering Supported: {0}", softwareEffectSupported), LogLevel.File);
-            vm.ViewModel.Model.Logger.Log(string.Format("RENDER : Maximum hardware texture size: {0}", maxTextureSize), LogLevel.File);
+            vm.ViewModel.Model.Logger.Log(string.Format("RENDER : Maximum hardware texture size: {0}", maxTextureSize), LogLevel.File); 
+        }
+
+        private void UnregisterEventHandlers()
+        {
+            var vm = DataContext as IWatchViewModel;
+            if (vm == null) return;
 
+            vm.VisualizationManager.RenderComplete -= VisualizationManagerRenderComplete;
+            vm.VisualizationManager.ResultsReadyToVisualize -= VisualizationManager_ResultsReadyToVisualize;
+            vm.ViewModel.PropertyChanged -= ViewModel_PropertyChanged;
+
+            vm.VisualizationManager.RenderSelection -= VisualizationManager_RenderSelection;
+            vm.VisualizationManager.UpdateGeometryOnNodeDeletion -= VisualizationManager_UpdateGeometryOnNodeDeletion;
+            vm.VisualizationManager.InitializeGeomtery -= VisualizationManager_InitializeGeomtery;
+
+            CompositionTarget.Rendering -= CompositionTarget_Rendering;
+            vm.ViewModel.Model.ShutdownStarted -= Model_ShutdownStarted;
+        }
+
+        private void RegisterEventHandlers(IWatchViewModel vm)
+        {
+            vm.VisualizationManager.RenderComplete += VisualizationManagerRenderComplete;
+            vm.VisualizationManager.ResultsReadyToVisualize += VisualizationManager_ResultsReadyToVisualize;
             vm.ViewModel.PropertyChanged += ViewModel_PropertyChanged;
+            vm.VisualizationManager.RenderSelection += VisualizationManager_RenderSelection;
+            vm.VisualizationManager.UpdateGeometryOnNodeDeletion += VisualizationManager_UpdateGeometryOnNodeDeletion;
+            vm.VisualizationManager.InitializeGeomtery += VisualizationManager_InitializeGeomtery;
+            CompositionTarget.Rendering += CompositionTarget_Rendering;
             vm.ViewModel.Model.ShutdownStarted += Model_ShutdownStarted;
-#if DEBUG
-            TestSelectionCommand = new Dynamo.UI.Commands.DelegateCommand(TestSelection, CanTestSelection);
-#endif
-            
+        }
+
+        /// <summary>
+        /// Initialize the Geometry everytime a workspace is opened or closed. 
+        /// Always, keep these DirectionalLight,Grid,Axes. These values are rendered
+        /// only once by helix, attaching them again will make no effect on helix 
+        /// </summary> 
+        private void VisualizationManager_InitializeGeomtery()
```

Comment:
Update these handler names in line with the proposed changes to event names above.